ci/build_setup.sh

```diff
@@ -56,6 +56,7 @@ function setup_gcc_toolchain() {
 }
 
 function setup_clang_toolchain() {
+  export CLANG_TOOLCHAIN_SETUP=1
```

Comment:
how about add a check to the CLANG_TOOLCHAIN_SETUP directly here to avoid possible repeated setup?

```
if [[ -n "$CLANG_TOOLCHAIN_SETUP" ]]; then
  return
fi
export CLANG_TOOLCHAIN_SETUP=1
```